// enhanced-ms/dist/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  Time: () => Time,
  createMs: () => createMs,
  default: () => index_default,
  formatMilliseconds: () => formatMilliseconds,
  getLanguage: () => getLanguage,
  languages: () => languages,
  ms: () => ms,
  parseDuration: () => parseDuration,
  parseMilliseconds: () => parseMilliseconds,
  parseUnits: () => parseUnits,
  units: () => units
});
module.exports = __toCommonJS(index_exports);

// src/time.ts
var Time = {
  Nanosecond: 1e-6,
  Microsecond: 1e-3,
  Millisecond: 1,
  Second: 1e3,
  Minute: 6e4,
  Hour: 36e5,
  Day: 864e5,
  // 24 hours
  Week: 6048e5,
  // 7 days
  Month: 2628e6,
  // 28 days
  Year: 31536e6,
  // 365 days
  Decade: 31536e7,
  // 10 years
  Century: 31536e8,
  // 100 years
  Millennium: 31536e9
  // 1000 years
};
var units = {
  nanosecond: 1e-6,
  microsecond: 1e-3,
  millisecond: 1,
  second: 1e3,
  minute: 6e4,
  hour: 36e5,
  day: 864e5,
  // 24 hours
  week: 6048e5,
  // 7 days
  month: 2628e6,
  // 28 days
  year: 31536e6,
  // 365 days
  decade: 31536e7,
  // 10 years
  century: 31536e8,
  // 100 years
  millennium: 31536e9
  // 1000 years
};

// src/languages/cs.ts
var UnitsMap = {
  nanosekundy: ["nanosekunda", "nanosekundy", "nanosekund"],
  mikrosekundy: ["mikrosekunda", "mikrosekundy", "mikrosekund"],
  milisekundy: ["milisekunda", "milisekundy", "milisekund"],
  sekundy: ["sekunda", "sekundy", "sekund"],
  minuty: ["minuta", "minuty", "minut"],
  hodiny: ["hodina", "hodiny", "hodin"],
  dny: ["den", "dny", "dn\xED"],
  t\u00FDdny: ["t\xFDden", "t\xFDdny", "t\xFDdn\u016F"],
  m\u011Bs\u00EDce: ["m\u011Bs\xEDc", "m\u011Bs\xEDce", "m\u011Bs\xEDc\u016F"],
  roky: ["rok", "roky", "let"],
  dek\u00E1dy: ["dek\xE1da", "dek\xE1dy", "dek\xE1d"],
  stolet\u00ED: ["stolet\xED", "stolet\xED", "stolet\xED"],
  tis\u00EDcilet\u00ED: ["tis\xEDcilet\xED", "tis\xEDcilet\xED", "tis\xEDcilet\xED"]
};
function formatTime(unit, count) {
  const words = UnitsMap[unit];
  let word;
  if (count === 1) word = words[0];
  else if (count >= 2 && count <= 4) word = words[1];
  else word = words[2];
  return word || "";
}
var cs_default = {
  decimal: ",",
  and: "a",
  units: {
    nanosecond: {
      name: (c) => formatTime("nanosekundy", c),
      abbreviation: "ns",
      matches: ["ns", "nanosekunda", "nanosekundy", "nanosekund"]
    },
    microsecond: {
      name: (c) => formatTime("mikrosekundy", c),
      abbreviation: "\u03BCs",
      matches: ["\u03BCs", "us", "mikrosekunda", "mikrosekundy", "mikrosekund"]
    },
    millisecond: {
      name: (c) => formatTime("milisekundy", c),
      abbreviation: "ms",
      matches: ["ms", "milisekunda", "milisekundy", "milisekund"]
    },
    second: {
      name: (c) => formatTime("sekundy", c),
      abbreviation: "s",
      matches: ["s", "sek", "sekunda", "sekundy", "sekund"]
    },
    minute: {
      name: (c) => formatTime("minuty", c),
      abbreviation: "min",
      matches: ["m", "min", "minuta", "minuty", "minut"]
    },
    hour: {
      name: (c) => formatTime("hodiny", c),
      abbreviation: "h",
      matches: ["h", "hod", "hodina", "hodiny", "hodin"]
    },
    day: {
      name: (c) => formatTime("dny", c),
      abbreviation: "d",
      matches: ["d", "den", "dny", "dn\xED"]
    },
    week: {
      name: (c) => formatTime("t\xFDdny", c),
      abbreviation: "t\xFDd.",
      matches: ["w", "t\xFDd", "t\xFDden", "t\xFDdny", "t\xFDdn\u016F"]
    },
    month: {
      name: (c) => formatTime("m\u011Bs\xEDce", c),
      abbreviation: "m\u011Bs.",
      matches: ["mo", "m\u011Bs", "m\u011Bs\xEDc", "m\u011Bs\xEDce", "m\u011Bs\xEDc\u016F"]
    },
    year: {
      name: (c) => formatTime("roky", c),
      abbreviation: "r.",
      matches: ["y", "r", "rok", "roky", "let"]
    },
    decade: {
      name: (c) => formatTime("dek\xE1dy", c),
      abbreviation: "dek.",
      matches: ["dek", "dek\xE1da", "dek\xE1dy", "dek\xE1d"]
    },
    century: {
      name: (c) => formatTime("stolet\xED", c),
      abbreviation: "stol.",
      matches: ["c", "stol", "stolet\xED"]
    },
    millennium: {
      name: (c) => formatTime("tis\xEDcilet\xED", c),
      abbreviation: "tis.",
      matches: ["mil", "tis", "tis\xEDcilet\xED"]
    }
  }
};

// src/languages/de.ts
var de_default = {
  decimal: ",",
  and: "und",
  units: {
    nanosecond: {
      name: (c) => c === 1 ? "Nanosekunde" : "Nanosekunden",
      abbreviation: "ns",
      matches: ["ns", "nanos", "nanosekunde", "nanosekunden"]
    },
    microsecond: {
      name: (c) => c === 1 ? "Mikrosekunde" : "Mikrosekunden",
      abbreviation: "\u03BCs",
      matches: ["us", "mk", "mikrosekunde", "mikrosekunden", "mikros"]
    },
    millisecond: {
      name: (c) => c === 1 ? "Millisekunde" : "Millisekunden",
      abbreviation: "ms",
      matches: ["ms", "ml", "mi", "millisekunde", "millisekunden", "millis"]
    },
    second: {
      name: (c) => c === 1 ? "Sekunde" : "Sekunden",
      abbreviation: "Sek.",
      matches: ["s", "sekunde", "sekunden", "sek", "seks"]
    },
    minute: {
      name: (c) => c === 1 ? "Minute" : "Minuten",
      abbreviation: "Min.",
      matches: ["m", "minute", "minuten", "min", "mins"]
    },
    hour: {
      name: (c) => c === 1 ? "Stunde" : "Stunden",
      abbreviation: "Std.",
      matches: ["h", "stunde", "stunden", "st"]
    },
    day: {
      name: (c) => c === 1 ? "Tag" : "Tage",
      abbreviation: "T.",
      matches: ["d", "t", "tag", "tage"]
    },
    week: {
      name: (c) => c === 1 ? "Woche" : "Wochen",
      abbreviation: "W.",
      matches: ["w", "wo", "woche", "wochen"]
    },
    month: {
      name: (c) => c === 1 ? "Monat" : "Monate",
      abbreviation: "M.",
      matches: ["mo", "monat", "monate"]
    },
    year: {
      name: (c) => c === 1 ? "Jahr" : "Jahre",
      abbreviation: "J.",
      matches: ["y", "a", "j", "jahr", "jahre"]
    },
    decade: {
      name: (c) => c === 1 ? "Jahrzehnt" : "Jahrzehnte",
      abbreviation: "Jz.",
      matches: ["jz", "jahrzehnt", "jahrzehnte"]
    },
    century: {
      name: (c) => c === 1 ? "Jahrhundert" : "Jahrhunderte",
      abbreviation: "Jh.",
      matches: ["c", "jh", "jhd", "jahrhundert", "jahrhunderte"]
    },
    millennium: {
      name: (c) => c === 1 ? "Jahrtausend" : "Jahrtausende",
      abbreviation: "Jt.",
      matches: ["jt", "jtd", "jtsd", "jahrt", "jahrtausend", "jahrtausende"]
    }
  }
};

// src/languages/en.ts
var en_default = {
  decimal: ".",
  and: "and",
  units: {
    nanosecond: {
      name: (c) => c === 1 ? "nanosecond" : "nanoseconds",
      abbreviation: "ns",
      matches: ["nanosecond", "nanoseconds", "ns"]
    },
    microsecond: {
      name: (c) => c === 1 ? "microsecond" : "microseconds",
      abbreviation: "\u03BCs",
      matches: ["microsecond", "microseconds", "\u03BCs"]
    },
    millisecond: {
      name: (c) => c === 1 ? "millisecond" : "milliseconds",
      abbreviation: "ms",
      matches: ["millisecond", "milliseconds", "ms"]
    },
    second: {
      name: (c) => c === 1 ? "second" : "seconds",
      abbreviation: "s",
      matches: ["second", "seconds", "s"]
    },
    minute: {
      name: (c) => c === 1 ? "minute" : "minutes",
      abbreviation: "m",
      matches: ["minute", "minutes", "m"]
    },
    hour: {
      name: (c) => c === 1 ? "hour" : "hours",
      abbreviation: "h",
      matches: ["hour", "hours", "h"]
    },
    day: {
      name: (c) => c === 1 ? "day" : "days",
      abbreviation: "d",
      matches: ["day", "days", "d"]
    },
    week: {
      name: (c) => c === 1 ? "week" : "weeks",
      abbreviation: "w",
      matches: ["week", "weeks", "w"]
    },
    month: {
      name: (c) => c === 1 ? "month" : "months",
      abbreviation: "mo",
      matches: ["month", "months", "mo"]
    },
    year: {
      name: (c) => c === 1 ? "year" : "years",
      abbreviation: "y",
      matches: ["year", "years", "y"]
    },
    decade: {
      name: (c) => c === 1 ? "decade" : "decades",
      abbreviation: "dec",
      matches: ["decade", "decades", "dec"]
    },
    century: {
      name: (c) => c === 1 ? "century" : "centuries",
      abbreviation: "c",
      matches: ["century", "centuries", "c"]
    },
    millennium: {
      name: (c) => c === 1 ? "millennium" : "millennia",
      abbreviation: "mil",
      matches: ["millennium", "millennia", "mil"]
    }
  }
};

// src/languages/es.ts
var es_default = {
  decimal: ",",
  and: "y",
  units: {
    nanosecond: {
      name: (c) => c === 1 ? "nanosegundo" : "nanosegundos",
      abbreviation: "ns",
      matches: ["nanosegundo", "nanosegundos", "ns"]
    },
    microsecond: {
      name: (c) => c === 1 ? "microsegundo" : "microsegundos",
      abbreviation: "\u03BCs",
      matches: ["microsegundo", "microsegundos", "\u03BCs"]
    },
    millisecond: {
      name: (c) => c === 1 ? "milisegundo" : "milisegundos",
      abbreviation: "ms",
      matches: ["milisegundo", "milisegundos", "ms"]
    },
    second: {
      name: (c) => c === 1 ? "segundo" : "segundos",
      abbreviation: "s",
      matches: ["segundo", "segundos", "s"]
    },
    minute: {
      name: (c) => c === 1 ? "minuto" : "minutos",
      abbreviation: "min",
      matches: ["minuto", "minutos", "min"]
    },
    hour: {
      name: (c) => c === 1 ? "hora" : "horas",
      abbreviation: "h",
      matches: ["hora", "horas", "h"]
    },
    day: {
      name: (c) => c === 1 ? "d\xEDa" : "d\xEDas",
      abbreviation: "d",
      matches: ["d\xEDa", "d\xEDas", "d"]
    },
    week: {
      name: (c) => c === 1 ? "semana" : "semanas",
      abbreviation: "sem",
      matches: ["semana", "semanas", "sem"]
    },
    month: {
      name: (c) => c === 1 ? "mes" : "meses",
      abbreviation: "mes",
      matches: ["mes", "meses", "mes"]
    },
    year: {
      name: (c) => c === 1 ? "a\xF1o" : "a\xF1os",
      abbreviation: "a",
      matches: ["a\xF1o", "a\xF1os", "a"]
    },
    decade: {
      name: (c) => c === 1 ? "d\xE9cada" : "d\xE9cadas",
      abbreviation: "d\xE9c",
      matches: ["d\xE9cada", "d\xE9cadas", "d\xE9c"]
    },
    century: {
      name: (c) => c === 1 ? "siglo" : "siglos",
      abbreviation: "sig",
      matches: ["siglo", "siglos", "sig"]
    },
    millennium: {
      name: (c) => c === 1 ? "milenio" : "milenios",
      abbreviation: "mil",
      matches: ["milenio", "milenios", "mil"]
    }
  }
};

// src/languages/fr.ts
var fr_default = {
  decimal: ",",
  and: "et",
  units: {
    nanosecond: {
      name: (c) => c === 1 ? "nanoseconde" : "nanosecondes",
      abbreviation: "ns",
      matches: ["nanoseconde", "nanosecondes", "ns"]
    },
    microsecond: {
      name: (c) => c === 1 ? "microseconde" : "microsecondes",
      abbreviation: "\u03BCs",
      matches: ["microseconde", "microsecondes", "\u03BCs"]
    },
    millisecond: {
      name: (c) => c === 1 ? "milliseconde" : "millisecondes",
      abbreviation: "ms",
      matches: ["milliseconde", "millisecondes", "ms"]
    },
    second: {
      name: (c) => c === 1 ? "seconde" : "secondes",
      abbreviation: "s",
      matches: ["seconde", "secondes", "s"]
    },
    minute: {
      name: (c) => c === 1 ? "minute" : "minutes",
      abbreviation: "min",
      matches: ["minute", "minutes", "min"]
    },
    hour: {
      name: (c) => c === 1 ? "heure" : "heures",
      abbreviation: "h",
      matches: ["heure", "heures", "h"]
    },
    day: {
      name: (c) => c === 1 ? "jour" : "jours",
      abbreviation: "j",
      matches: ["jour", "jours", "j"]
    },
    week: {
      name: (c) => c === 1 ? "semaine" : "semaines",
      abbreviation: "sem",
      matches: ["semaine", "semaines", "sem"]
    },
    month: {
      name: (_c) => "mois",
      // In French, the singular and plural form is identical.
      abbreviation: "mo",
      matches: ["mois", "mo"]
    },
    year: {
      name: (c) => c === 1 ? "an" : "ans",
      abbreviation: "an",
      matches: ["an", "ans"]
    },
    decade: {
      name: (c) => c === 1 ? "d\xE9cennie" : "d\xE9cennies",
      abbreviation: "dec",
      matches: ["d\xE9cennie", "d\xE9cennies", "dec"]
    },
    century: {
      name: (c) => c === 1 ? "si\xE8cle" : "si\xE8cles",
      abbreviation: "si\xE8",
      matches: ["si\xE8cle", "si\xE8cles", "si\xE8"]
    },
    millennium: {
      name: (c) => c === 1 ? "mill\xE9naire" : "mill\xE9naires",
      abbreviation: "mil",
      matches: ["mill\xE9naire", "mill\xE9naires", "mil"]
    }
  }
};

// src/languages/it.ts
var it_default = {
  decimal: ",",
  and: "e",
  units: {
    nanosecond: {
      name: (c) => c === 1 ? "nanosecondo" : "nanosecondi",
      abbreviation: "ns",
      matches: ["nanosecondo", "nanosecondi", "ns"]
    },
    microsecond: {
      name: (c) => c === 1 ? "microsecondo" : "microsecondi",
      abbreviation: "\u03BCs",
      matches: ["microsecondo", "microsecondi", "\u03BCs"]
    },
    millisecond: {
      name: (c) => c === 1 ? "millisecondo" : "millisecondi",
      abbreviation: "ms",
      matches: ["millisecondo", "millisecondi", "ms"]
    },
    second: {
      name: (c) => c === 1 ? "secondo" : "secondi",
      abbreviation: "s",
      matches: ["secondo", "secondi", "s"]
    },
    minute: {
      name: (c) => c === 1 ? "minuto" : "minuti",
      abbreviation: "min",
      matches: ["minuto", "minuti", "min"]
    },
    hour: {
      name: (c) => c === 1 ? "ora" : "ore",
      abbreviation: "h",
      matches: ["ora", "ore", "h"]
    },
    day: {
      name: (c) => c === 1 ? "giorno" : "giorni",
      abbreviation: "g",
      matches: ["giorno", "giorni", "g"]
    },
    week: {
      name: (c) => c === 1 ? "settimana" : "settimane",
      abbreviation: "sett",
      matches: ["settimana", "settimane", "sett"]
    },
    month: {
      name: (c) => c === 1 ? "mese" : "mesi",
      abbreviation: "mes",
      matches: ["mese", "mesi", "mes"]
    },
    year: {
      name: (c) => c === 1 ? "anno" : "anni",
      abbreviation: "a",
      matches: ["anno", "anni", "a"]
    },
    decade: {
      name: (c) => c === 1 ? "decennio" : "decenni",
      abbreviation: "dec",
      matches: ["decennio", "decenni", "dec"]
    },
    century: {
      name: (c) => c === 1 ? "secolo" : "secoli",
      abbreviation: "sc",
      matches: ["secolo", "secoli", "sc"]
    },
    millennium: {
      name: (c) => c === 1 ? "millennio" : "millenni",
      abbreviation: "mil",
      matches: ["millennio", "millenni", "mil"]
    }
  }
};

// src/languages/mi.ts
var mi_default = {
  decimal: ".",
  and: "me te",
  units: {
    nanosecond: {
      name: "h\u0113kona nanomiri",
      matches: [
        "h\u0113kona nanomiri",
        "h\u0113konananomiri",
        "hekona nanomiri",
        "hekonananomiri"
      ]
    },
    microsecond: {
      name: "h\u0113kona miriona",
      matches: [
        "h\u0113kona miriona",
        "h\u0113konamiriona",
        "hekona miriona",
        "hekonamiriona"
      ]
    },
    millisecond: {
      name: "h\u0113kona miri",
      matches: ["h\u0113kona miri", "k\u0101konamiri", "hekona miri", "hekonamiri"]
    },
    second: {
      name: "h\u0113kona",
      matches: ["h\u0113kona", "hekona"]
    },
    minute: {
      name: "meneti",
      matches: ["meneti"]
    },
    hour: {
      name: "h\u0101ora",
      matches: ["h\u0101ora", "haora"]
    },
    day: {
      name: "r\u0101",
      matches: ["r\u0101", "ra"]
    },
    week: {
      name: "wiki",
      matches: ["wiki"]
    },
    month: {
      name: "marama",
      matches: ["marama"]
    },
    year: {
      name: "tau",
      matches: ["tau", "houanga"]
    },
    decade: {
      name: "tekau tau",
      matches: ["tekautau", "tekau tau"]
    },
    century: {
      name: "rau tau",
      matches: ["rautau", "rau tau"]
    },
    millennium: {
      name: "mano tau",
      matches: ["manotau", "mano tau"]
    }
  }
};

// src/languages/nl.ts
var nl_default = {
  decimal: ",",
  and: "en",
  units: {
    nanosecond: {
      name: (c) => c === 1 ? "nanoseconde" : "nanoseconden",
      abbreviation: "ns",
      matches: ["nanoseconde", "nanoseconden", "ns"]
    },
    microsecond: {
      name: (c) => c === 1 ? "microseconde" : "microseconden",
      abbreviation: "\u03BCs",
      matches: ["microseconde", "microseconden", "\u03BCs"]
    },
    millisecond: {
      name: (c) => c === 1 ? "milliseconde" : "milliseconden",
      abbreviation: "ms",
      matches: ["milliseconde", "milliseconden", "ms"]
    },
    second: {
      name: (c) => c === 1 ? "seconde" : "seconden",
      abbreviation: "s",
      matches: ["seconde", "seconden", "s"]
    },
    minute: {
      name: (c) => c === 1 ? "minuut" : "minuten",
      abbreviation: "min",
      matches: ["minuut", "minuten", "min"]
    },
    hour: {
      name: (c) => c === 1 ? "uur" : "uren",
      abbreviation: "h",
      matches: ["uur", "uren", "h"]
    },
    day: {
      name: (c) => c === 1 ? "dag" : "dagen",
      abbreviation: "d",
      matches: ["dag", "dagen", "d"]
    },
    week: {
      name: (c) => c === 1 ? "week" : "weken",
      abbreviation: "w",
      matches: ["week", "weken", "w"]
    },
    month: {
      name: (c) => c === 1 ? "maand" : "maanden",
      abbreviation: "mnd",
      matches: ["maand", "maanden", "mnd"]
    },
    year: {
      name: (c) => c === 1 ? "jaar" : "jaren",
      abbreviation: "jr",
      matches: ["jaar", "jaren", "jr"]
    },
    decade: {
      name: (c) => c === 1 ? "decennium" : "decennia",
      abbreviation: "dec",
      matches: ["decennium", "decennia", "dec"]
    },
    century: {
      name: (c) => c === 1 ? "eeuw" : "eeuwen",
      abbreviation: "eeuw",
      matches: ["eeuw", "eeuwen"]
    },
    millennium: {
      name: (c) => c === 1 ? "millennium" : "millennia",
      abbreviation: "mil",
      matches: ["millennium", "millennia", "mil"]
    }
  }
};

// src/languages/pl.ts
var UnitsMap2 = {
  nanosekundy: ["nanosekunda", "nanosekundy", "nanosekund"],
  mikrosekundy: ["mikrosekunda", "mikrosekundy", "mikrosekund"],
  milisekundy: ["milisekunda", "milisekundy", "milisekund"],
  sekundy: ["sekunda", "sekundy", "sekund"],
  minuty: ["minuta", "minuty", "minut"],
  godziny: ["godzina", "godziny", "godzin"],
  dni: ["dzie\u0144", "dni", "dni"],
  tygodnie: ["tydzie\u0144", "tygodnie", "tygodni"],
  miesi\u0105ce: ["miesi\u0105c", "miesi\u0105ce", "miesi\u0119cy"],
  lata: ["rok", "lata", "lat"],
  dekady: ["dekada", "dekady", "dekad"],
  wieki: ["wiek", "wieki", "wiek\xF3w"],
  tysi\u0105clecia: ["tysi\u0105clecie", "tysi\u0105clecia", "tysi\u0105cleci"]
};
function formatTime2(unit, count) {
  const words = UnitsMap2[unit];
  const lastDigit = count % 10;
  const lastTwoDigits = count % 100;
  let word;
  if (lastTwoDigits >= 11 && lastTwoDigits <= 19) word = words[2];
  else if (lastDigit === 1) word = words[0];
  else if (lastDigit >= 2 && lastDigit <= 4) word = words[1];
  else word = words[2];
  return word || "";
}
var pl_default = {
  decimal: ",",
  and: "i",
  units: {
    nanosecond: {
      name: (c) => formatTime2("nanosekundy", c),
      abbreviation: "ns",
      matches: ["ns", "nanosekunda", "nanosekundy", "nanosekund"]
    },
    microsecond: {
      name: (c) => formatTime2("mikrosekundy", c),
      abbreviation: "\u03BCs",
      matches: ["\u03BCs", "us", "mikrosekunda", "mikrosekundy", "mikrosekund"]
    },
    millisecond: {
      name: (c) => formatTime2("milisekundy", c),
      abbreviation: "ms",
      matches: ["ms", "milisekunda", "milisekundy", "milisekund"]
    },
    second: {
      name: (c) => formatTime2("sekundy", c),
      abbreviation: "s",
      matches: ["s", "sek", "sekunda", "sekundy", "sekund"]
    },
    minute: {
      name: (c) => formatTime2("minuty", c),
      abbreviation: "min",
      matches: ["m", "min", "minuta", "minuty", "minut"]
    },
    hour: {
      name: (c) => formatTime2("godziny", c),
      abbreviation: "godz.",
      matches: ["h", "godz", "godzina", "godziny", "godzin"]
    },
    day: {
      name: (c) => formatTime2("dni", c),
      abbreviation: "d",
      matches: ["d", "dzie\u0144", "dni"]
    },
    week: {
      name: (c) => formatTime2("tygodnie", c),
      abbreviation: "tydz.",
      matches: ["w", "tydz", "tydzie\u0144", "tygodnie", "tygodni"]
    },
    month: {
      name: (c) => formatTime2("miesi\u0105ce", c),
      abbreviation: "mies.",
      matches: ["mo", "mies", "miesi\u0105c", "miesi\u0105ce", "miesi\u0119cy"]
    },
    year: {
      name: (c) => formatTime2("lata", c),
      abbreviation: "r.",
      matches: ["y", "r", "rok", "lata", "lat"]
    },
    decade: {
      name: (c) => formatTime2("dekady", c),
      abbreviation: "dek.",
      matches: ["dek", "dekada", "dekady", "dekad"]
    },
    century: {
      name: (c) => formatTime2("wieki", c),
      abbreviation: "w.",
      matches: ["c", "wiek", "wieki", "wiek\xF3w"]
    },
    millennium: {
      name: (c) => formatTime2("tysi\u0105clecia", c),
      abbreviation: "tys.",
      matches: ["mil", "tys", "tysi\u0105clecie", "tysi\u0105clecia", "tysi\u0105cleci"]
    }
  }
};

// src/languages/ru.ts
var UnitsMap3 = {
  \u043D\u0430\u043D\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u044B: ["\u043D\u0430\u043D\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u0430", "\u043D\u0430\u043D\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u044B", "\u043D\u0430\u043D\u043E\u0441\u0435\u043A\u0443\u043D\u0434"],
  \u043C\u0438\u043A\u0440\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u044B: ["\u043C\u0438\u043A\u0440\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u0430", "\u043C\u0438\u043A\u0440\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u044B", "\u043C\u0438\u043A\u0440\u043E\u0441\u0435\u043A\u0443\u043D\u0434"],
  \u043C\u0438\u043B\u043B\u0438\u0441\u0435\u043A\u0443\u043D\u0434\u044B: ["\u043C\u0438\u043B\u043B\u0438\u0441\u0435\u043A\u0443\u043D\u0434\u0430", "\u043C\u0438\u043B\u043B\u0438\u0441\u0435\u043A\u0443\u043D\u0434\u044B", "\u043C\u0438\u043B\u043B\u0438\u0441\u0435\u043A\u0443\u043D\u0434"],
  \u0441\u0435\u043A\u0443\u043D\u0434\u044B: ["\u0441\u0435\u043A\u0443\u043D\u0434\u0430", "\u0441\u0435\u043A\u0443\u043D\u0434\u044B", "\u0441\u0435\u043A\u0443\u043D\u0434"],
  \u043C\u0438\u043D\u0443\u0442\u044B: ["\u043C\u0438\u043D\u0443\u0442\u0430", "\u043C\u0438\u043D\u0443\u0442\u044B", "\u043C\u0438\u043D\u0443\u0442"],
  \u0447\u0430\u0441\u044B: ["\u0447\u0430\u0441", "\u0447\u0430\u0441\u0430", "\u0447\u0430\u0441\u043E\u0432"],
  \u0434\u043D\u0438: ["\u0434\u0435\u043D\u044C", "\u0434\u043D\u044F", "\u0434\u043D\u0435\u0439"],
  \u043D\u0435\u0434\u0435\u043B\u0438: ["\u043D\u0435\u0434\u0435\u043B\u044F", "\u043D\u0435\u0434\u0435\u043B\u0438", "\u043D\u0435\u0434\u0435\u043B\u044C"],
  \u043C\u0435\u0441\u044F\u0446\u044B: ["\u043C\u0435\u0441\u044F\u0446", "\u043C\u0435\u0441\u044F\u0446\u0430", "\u043C\u0435\u0441\u044F\u0446\u0435\u0432"],
  \u0433\u043E\u0434\u044B: ["\u0433\u043E\u0434", "\u0433\u043E\u0434\u0430", "\u043B\u0435\u0442"],
  \u0434\u0435\u0441\u044F\u0442\u0438\u043B\u0435\u0442\u0438\u044F: ["\u0434\u0435\u0441\u044F\u0442\u0438\u043B\u0435\u0442\u0438\u0435", "\u0434\u0435\u0441\u044F\u0442\u0438\u043B\u0435\u0442\u0438\u044F", "\u0434\u0435\u0441\u044F\u0442\u0438\u043B\u0435\u0442\u0438\u0439"],
  \u0432\u0435\u043A\u0430: ["\u0432\u0435\u043A", "\u0432\u0435\u043A\u0430", "\u0432\u0435\u043A\u043E\u0432"],
  \u0442\u044B\u0441\u044F\u0447\u0435\u043B\u0435\u0442\u0438\u044F: ["\u0442\u044B\u0441\u044F\u0447\u0435\u043B\u0435\u0442\u0438\u0435", "\u0442\u044B\u0441\u044F\u0447\u0435\u043B\u0435\u0442\u0438\u044F", "\u0442\u044B\u0441\u044F\u0447\u0435\u043B\u0435\u0442\u0438\u0439"]
};
function formatTime3(unit, count) {
  const words = UnitsMap3[unit];
  const lastDigit = count % 10;
  const lastTwoDigits = count % 100;
  let word;
  if (lastTwoDigits >= 11 && lastTwoDigits <= 19) word = words[2];
  else if (lastDigit === 1) word = words[0];
  else if (lastDigit >= 2 && lastDigit <= 4) word = words[1];
  else word = words[2];
  return word || "";
}
var ru_default = {
  decimal: ".",
  and: "\u0438",
  units: {
    nanosecond: {
      name: (c) => formatTime3("\u043D\u0430\u043D\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u044B", c),
      abbreviation: "\u043D\u0441",
      matches: ["\u043D\u0441", "\u043D\u0430\u043D\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u0430", "\u043D\u0430\u043D\u043E\u0441\u0435\u043A\u0443\u043D\u0434", "\u043D\u0430\u043D\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u044B"]
    },
    microsecond: {
      name: (c) => formatTime3("\u043C\u0438\u043A\u0440\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u044B", c),
      abbreviation: "\u043C\u043A\u0441",
      matches: ["\u043C\u043A\u0441", "\u043C\u0438\u043A\u0440\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u0430", "\u043C\u0438\u043A\u0440\u043E\u0441\u0435\u043A\u0443\u043D\u0434", "\u043C\u0438\u043A\u0440\u043E\u0441\u0435\u043A\u0443\u043D\u0434\u044B"]
    },
    millisecond: {
      name: (c) => formatTime3("\u043C\u0438\u043B\u043B\u0438\u0441\u0435\u043A\u0443\u043D\u0434\u044B", c),
      abbreviation: "\u043C\u0441",
      matches: ["\u043C\u0441", "\u043C\u0438\u043B\u043B\u0438\u0441\u0435\u043A\u0443\u043D\u0434\u0430", "\u043C\u0438\u043B\u043B\u0438\u0441\u0435\u043A\u0443\u043D\u0434", "\u043C\u0438\u043B\u043B\u0438\u0441\u0435\u043A\u0443\u043D\u0434\u044B", "\u043C\u0441\u0435\u043A"]
    },
    second: {
      name: (c) => formatTime3("\u0441\u0435\u043A\u0443\u043D\u0434\u044B", c),
      abbreviation: "\u0441",
      matches: ["\u0441", "\u0441\u0435\u043A\u0443\u043D\u0434\u0430", "\u0441\u0435\u043A\u0443\u043D\u0434", "\u0441\u0435\u043A\u0443\u043D\u0434\u044B", "\u0441\u0435\u043A"]
    },
    minute: {
      name: (c) => formatTime3("\u043C\u0438\u043D\u0443\u0442\u044B", c),
      abbreviation: "\u043C",
      matches: ["\u043C", "\u043C\u0438\u043D\u0443\u0442\u0430", "\u043C\u0438\u043D\u0443\u0442\u044B", "\u043C\u0438\u043D\u0443\u0442", "\u043C\u0438\u043D"]
    },
    hour: {
      name: (c) => formatTime3("\u0447\u0430\u0441\u044B", c),
      abbreviation: "\u0447",
      matches: ["\u0447", "\u0447\u0430\u0441", "\u0447\u0430\u0441\u0430", "\u0447\u0430\u0441\u043E\u0432"]
    },
    day: {
      name: (c) => formatTime3("\u0434\u043D\u0438", c),
      abbreviation: "\u0434",
      matches: ["\u0434", "\u0434\u0435\u043D\u044C", "\u0434\u043D\u044F", "\u0434\u043D\u0435\u0439"]
    },
    week: {
      name: (c) => formatTime3("\u043D\u0435\u0434\u0435\u043B\u0438", c),
      abbreviation: "\u043D",
      matches: ["\u043D", "\u043D\u0435\u0434\u0435\u043B\u044F", "\u043D\u0435\u0434\u0435\u043B\u0438", "\u043D\u0435\u0434\u0435\u043B\u044C"]
    },
    month: {
      name: (c) => formatTime3("\u043C\u0435\u0441\u044F\u0446\u044B", c),
      abbreviation: "\u043C\u0435\u0441",
      matches: ["\u043C\u0435\u0441", "\u043C\u0435\u0441\u044F\u0446", "\u043C\u0435\u0441\u044F\u0446\u0430", "\u043C\u0435\u0441\u044F\u0446\u0435\u0432"]
    },
    year: {
      name: (c) => formatTime3("\u0433\u043E\u0434\u044B", c),
      abbreviation: "\u0433",
      matches: ["\u0433", "\u0433\u043E\u0434", "\u0433\u043E\u0434\u0430", "\u043B\u0435\u0442", "\u043B"]
    },
    decade: {
      name: (c) => formatTime3("\u0434\u0435\u0441\u044F\u0442\u0438\u043B\u0435\u0442\u0438\u044F", c),
      abbreviation: "\u0434\u0441\u043B",
      matches: [
        "\u0434\u0435\u0441\u044F\u0442\u0438\u043B\u0435\u0442\u0438\u0435",
        "\u0434\u0435\u0441\u044F\u0442\u0438\u043B\u0435\u0442\u0438\u044F",
        "\u0434\u0435\u0441\u044F\u0442\u0438\u043B\u0435\u0442\u0438\u0439",
        "\u0434\u0435\u0441\u044F\u0442\u043A\u043E\u0432 \u043B\u0435\u0442",
        "\u0434\u0435\u0441\u044F\u0442\u043A\u0430 \u043B\u0435\u0442",
        "\u0434\u0435\u0441\u044F\u0442\u043E\u043A \u043B\u0435\u0442"
      ]
    },
    century: {
      name: (c) => formatTime3("\u0432\u0435\u043A\u0430", c),
      abbreviation: "\u0432\u043A",
      matches: ["\u0432\u0435\u043A", "\u0432\u0435\u043A\u0430", "\u0432\u0435\u043A\u043E\u0432"]
    },
    millennium: {
      name: (c) => formatTime3("\u0442\u044B\u0441\u044F\u0447\u0435\u043B\u0435\u0442\u0438\u044F", c),
      abbreviation: "\u0442\u044B\u0441",
      matches: [
        "\u0442\u044B\u0441",
        "\u0442\u044B\u0441\u044F\u0447\u0435\u043B\u0435\u0442\u0438\u0435",
        "\u0442\u044B\u0441\u044F\u0447\u0430\u043B\u0435\u0442\u0438\u044F",
        "\u0442\u044B\u0441\u044F\u0447\u0430\u043B\u0435\u0442\u0438\u0439",
        "\u0442\u044B\u0441\u044F\u0447\u0430 \u043B\u0435\u0442",
        "\u0442\u044B\u0441\u044F\u0447\u0438 \u043B\u0435\u0442",
        "\u0442\u044B\u0441\u044F\u0447 \u043B\u0435\u0442"
      ]
    }
  }
};

// src/languages/zh-CN.ts
var zh_CN_default = {
  decimal: ".",
  and: "\u548C",
  units: {
    nanosecond: {
      name: "\u7EB3\u79D2",
      abbreviation: "ns",
      matches: ["\u7EB3\u79D2", "ns"]
    },
    microsecond: {
      name: "\u5FAE\u79D2",
      abbreviation: "\u03BCs",
      matches: ["\u5FAE\u79D2", "\u03BCs", "us"]
    },
    millisecond: {
      name: "\u6BEB\u79D2",
      abbreviation: "ms",
      matches: ["\u6BEB\u79D2", "ms"]
    },
    second: {
      name: "\u79D2",
      abbreviation: "s",
      matches: ["\u79D2", "s"]
    },
    minute: {
      name: "\u5206\u949F",
      abbreviation: "\u5206",
      matches: ["\u5206\u949F", "\u5206", "m", "min"]
    },
    hour: {
      name: "\u5C0F\u65F6",
      abbreviation: "\u65F6",
      matches: ["\u5C0F\u65F6", "\u65F6", "h", "hour"]
    },
    day: {
      name: "\u5929",
      abbreviation: "\u5929",
      matches: ["\u5929", "\u65E5", "d", "day"]
    },
    week: {
      name: "\u5468",
      abbreviation: "\u5468",
      matches: ["\u5468", "\u661F\u671F", "w", "week"]
    },
    month: {
      name: "\u4E2A\u6708",
      abbreviation: "\u6708",
      matches: ["\u4E2A\u6708", "\u6708", "mo", "month"]
    },
    year: {
      name: "\u5E74",
      abbreviation: "\u5E74",
      matches: ["\u5E74", "y", "year"]
    },
    decade: {
      name: "\u5341\u5E74",
      abbreviation: "\u5341\u5E74",
      matches: ["\u5341\u5E74", "dec", "decade"]
    },
    century: {
      name: "\u4E16\u7EAA",
      abbreviation: "\u4E16\u7EAA",
      matches: ["\u4E16\u7EAA", "c", "century"]
    },
    millennium: {
      name: "\u5343\u5E74",
      abbreviation: "\u5343\u5E74",
      matches: ["\u5343\u5E74", "mil", "millennium"]
    }
  }
};

// src/languages/index.ts
var languages = {
  cs: cs_default,
  de: de_default,
  en: en_default,
  es: es_default,
  fr: fr_default,
  it: it_default,
  mi: mi_default,
  nl: nl_default,
  pl: pl_default,
  ru: ru_default,
  "zh-CN": zh_CN_default
};

// src/languages/helpers/make-language.ts
function makeLanguage(languageDefinition) {
  const decimalSeparator = languageDefinition.decimal;
  const thousandSeparator = decimalSeparator === "." ? "," : ".";
  const matcherRegex = new RegExp(
    // biome-ignore lint/style/useTemplate: Better readability
    `(?![${decimalSeparator}${thousandSeparator}])[\\d${decimalSeparator}${thousandSeparator}]+|(?<=\\s|\\d)((?:-)?(` + // Units
    Object.values(languageDefinition.units).flatMap(({ matches }) => matches).sort((a, b) => b.length - a.length).join("|") + "))",
    "gi"
  );
  const timeUnits = Object.fromEntries(
    Object.entries(languageDefinition.units).flatMap(
      ([unit, definition]) => [unit, ...definition.matches].map((m) => [
        m,
        { ...definition, ms: units[unit] }
      ])
    )
  );
  return {
    andValue: languageDefinition.and,
    decimalSeparator,
    thousandSeparator,
    matcherRegex,
    timeUnits,
    supportsAbbreviations: Object.values(timeUnits).every((u) => u.abbreviation)
  };
}
var languageCache = /* @__PURE__ */ new Map();
function getLanguage(localeOrLanguageDefinition) {
  const languageDefinition = typeof localeOrLanguageDefinition === "string" ? languages[localeOrLanguageDefinition] : localeOrLanguageDefinition;
  if (!languageDefinition) throw new Error("Invalid language");
  const key = JSON.stringify(languageDefinition);
  if (languageCache.has(key)) return languageCache.get(key);
  const language = makeLanguage(languageDefinition);
  languageCache.set(key, language);
  return language;
}

// src/format/helpers/format-unit.ts
function formatUnit(unit, count, options) {
  const { useAbbreviations, hideUnitNames, minimumDigits } = options;
  const amount = count.toString().padStart(minimumDigits, "0");
  if (hideUnitNames) return amount;
  const name = pluraliseUnit(unit, count, options);
  return `${amount}${useAbbreviations ? "" : " "}${name}`;
}
function pluraliseUnit(unit, count, options) {
  const { useAbbreviations } = options;
  const factory = useAbbreviations && unit.abbreviation ? unit.abbreviation : unit.name;
  return typeof factory === "function" ? factory(Math.abs(count)) : factory;
}

// src/format/helpers/parse-milliseconds.ts
var parseUnits = {
  year: units.year,
  day: units.day,
  hour: units.hour,
  minute: units.minute,
  second: units.second,
  millisecond: units.millisecond,
  microsecond: units.microsecond,
  nanosecond: units.nanosecond
};
function parseMilliseconds(milliseconds, includedUnits = Object.keys(parseUnits)) {
  let remainingMs = milliseconds;
  const parsed = {};
  for (const key in parseUnits) {
    const unit = key;
    if (includedUnits.includes(unit)) {
      parsed[unit] = Math.trunc(remainingMs / parseUnits[unit]);
      remainingMs %= parseUnits[unit];
    } else {
      parsed[unit] = null;
    }
  }
  return parsed;
}

// src/format/helpers/resolve-options.ts
var defaultFormatOptions = {
  hideUnitNames: false,
  useAbbreviations: false,
  includeZero: false,
  includeMs: false,
  includeSubMs: false,
  includedUnits: ["year", "day", "hour", "minute", "second"],
  unitLimit: -1,
  unitSeparator: " ",
  minimumDigits: 0
};
function resolveFormatOptions(options) {
  let {
    hideUnitNames = defaultFormatOptions.hideUnitNames,
    useAbbreviations = defaultFormatOptions.useAbbreviations,
    includeZero = defaultFormatOptions.includeZero,
    includeMs = defaultFormatOptions.includeMs,
    includeSubMs = defaultFormatOptions.includeSubMs,
    includedUnits = [...defaultFormatOptions.includedUnits],
    unitLimit = defaultFormatOptions.unitLimit,
    unitSeparator = defaultFormatOptions.unitSeparator,
    minimumDigits = defaultFormatOptions.minimumDigits
  } = options;
  if (typeof includeMs !== "boolean")
    throw new Error("Invalid includeMs option");
  if (typeof includeSubMs !== "boolean")
    throw new Error("Invalid includeSubMs option");
  if (typeof hideUnitNames !== "boolean")
    throw new Error("Invalid hideUnitNames option");
  if (typeof useAbbreviations !== "boolean")
    throw new Error("Invalid useAbbreviations option");
  if (typeof includeZero !== "boolean")
    throw new Error("Invalid includeZero option");
  if (!Array.isArray(includedUnits))
    throw new Error("Invalid includedUnits option");
  if (typeof unitLimit !== "number")
    throw new Error("Invalid unitLimit option");
  if (typeof unitSeparator !== "string")
    throw new Error("Invalid unitSeparator option");
  if (typeof minimumDigits !== "number")
    throw new Error("Invalid minimumDigits option");
  if (includeSubMs) {
    if (includeMs === false)
      throw new Error(
        "Conflicting options: `includeSubMs` is true while `includeMs` is false."
      );
    includeMs = true;
  }
  if (includeMs) includedUnits.push("millisecond");
  if (includeSubMs) includedUnits.push("microsecond", "nanosecond");
  return {
    hideUnitNames,
    useAbbreviations,
    includeZero,
    includedUnits,
    unitLimit,
    unitSeparator,
    minimumDigits,
    __transformDuration__: options.__transformDuration__
  };
}
var formatOptionsPresets = {
  short: {
    useAbbreviations: true,
    unitLimit: 2
  },
  fullPrecision: {
    includeMs: true,
    includeSubMs: true
  },
  colonNotation: {
    hideUnitNames: true,
    unitLimit: 3,
    includeZero: true,
    includedUnits: ["hour", "minute", "second"],
    unitSeparator: ":",
    minimumDigits: 2,
    // I couldn't think of an option that would be better than this
    __transformDuration__: (duration) => duration.replace(/^00:/, "")
  }
};
function resolveFormatPresetOptions(options = {}) {
  if (typeof options === "string") options = { extends: options };
  return {
    ...options.extends && formatOptionsPresets[options.extends],
    ...options,
    preset: void 0
  };
}

// src/format/index.ts
function formatMilliseconds(milliseconds, language = getLanguage("en"), options = {}) {
  const presetOptions = resolveFormatPresetOptions(options);
  const resolvedOptions = resolveFormatOptions(presetOptions);
  const { includedUnits } = resolvedOptions;
  const time = parseMilliseconds(milliseconds, includedUnits);
  const entries = [];
  for (const key in time) {
    const unit = key;
    const value = time[unit];
    entries.push({ unit: language.timeUnits[unit], amount: value });
  }
  const { includeZero } = resolvedOptions;
  const filtered = entries.filter((entry) => {
    if (entry.amount === null) return false;
    if (entry.amount === 0) return includeZero;
    return true;
  });
  const { unitLimit } = resolvedOptions;
  const formatted = [];
  for (let i = 0; i < filtered.length; i++) {
    const entry = filtered.at(i);
    if (unitLimit !== -1 && formatted.length >= unitLimit) break;
    const format = formatUnit(entry.unit, entry.amount, resolvedOptions);
    formatted.push(format);
  }
  const { unitSeparator, __transformDuration__ } = resolvedOptions;
  const duration = formatted.join(unitSeparator) || null;
  return duration && __transformDuration__ ? __transformDuration__(duration) : duration;
}

// src/parse/index.ts
function parseDuration(duration, language = getLanguage("en")) {
  const matches = duration.toLowerCase().match(language.matcherRegex);
  if (!matches || matches.length === 0) return 0;
  let total = 0;
  for (let i = 0; i < matches.length; i += 2) {
    const [amount, unit] = matches.slice(i, i + 2);
    const ms2 = language.timeUnits[unit]?.ms;
    if (ms2) total += Number(amount) * ms2;
  }
  return total;
}

// src/factory.ts
function createMs(options = {}) {
  const language = getLanguage(options.language ?? "en");
  const defaultFormatOptions2 = (
    //
    resolveFormatPresetOptions(options.formatOptions)
  );
  function ms2(...args) {
    switch (typeof args[0]) {
      case "number": {
        const [milliseconds, additionalOptions] = args;
        if (Number.isNaN(milliseconds) || !Number.isFinite(milliseconds))
          throw new TypeError("Expected a finite number");
        if (milliseconds < 0) throw new TypeError("Expected a positive number");
        const presetOptions = resolveFormatPresetOptions(additionalOptions);
        const formatOptions = { ...defaultFormatOptions2, ...presetOptions };
        return formatMilliseconds(milliseconds, language, formatOptions);
      }
      case "string": {
        const [duration] = args;
        return parseDuration(duration, language) ?? 0;
      }
      default:
        throw new TypeError("Expected a finite number or a string");
    }
  }
  return ms2;
}

// src/index.ts
var ms = createMs({ language: "en" });
var index_default = ms;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Time,
  createMs,
  formatMilliseconds,
  getLanguage,
  languages,
  ms,
  parseDuration,
  parseMilliseconds,
  parseUnits,
  units
});
